### PR TITLE
Fixed file generator utility after code review

### PR DESCRIPTION
**Pull request by slavapa**
Please review.

### CODE REVIEW

File: lib/file_utils/file_generator/file_generator.rb [hunk 1]
```diff
@@ -1,140 +1,122 @@
-require 'fileutils'
-require_relative 'file_generator_parameters'
 require 'digest'
+require 'file_utils/file_generator/file_generator_parameters'
+require 'fileutils'
+require 'params'
```
Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator.rb]: Why did you add the library if you are not using it?

File: lib/file_utils/file_generator/file_generator.rb [hunk 2]
```diff
@@ -1,140 +1,122 @@
...
+class FileGenerator
+  attr_accessor :file_gen_params
+  attr_reader :one_kb_string, :one_mb_string
+
+  def initialize (file_gen_params = FileGeneratorParameters.new())
+    @file_gen_params = file_gen_params
+  end
+
+  def one_mb_string
+    @one_mb_string || @one_mb_string = prepare_one_mb_string
```
Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator.rb:16]: one_mb_string is same as one_kb string.
Make one method which gets a parameter.

File: lib/file_utils/file_generator/file_generator.rb [hunk 3]
```diff
@@ -1,140 +1,122 @@
...
+    "#{Time.new.to_i}_#{get_small_rand_unique_str}"
+  end
+
+  def get_random_letter
+    (rand(122-97) + 97).chr
+  end
+
+  def determine_dir_recursively (path)
+    if File.directory?(path)
+      #puts "The given path already exists #{path}"
```
Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator.rb]: change the whole if to one line:
FileUtils.mkdir_p path unless File.directory?(path)

delete this function if you use it only once.

File: lib/file_utils/file_generator/file_generator.rb (hunk 1 above)
Comment (slavapa) [inline lib/file_utils/file_generator/file_generator.rb]: I thought to use.
I'll use this or delete this in the next commit today. 

-----Original Message-----

Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator.rb]: ok, use the same pull request (i.e. just commit to the branch).

On Tue, May 22, 2012 at 9:12 AM, Slava <
reply@reply.github.com

Comment (slavapa) [inline lib/file_utils/file_generator/file_generator.rb]: Removed require 'params'

File: lib/file_utils/file_generator/file_generator.rb (hunk 3 above)
Comment (slavapa) [inline lib/file_utils/file_generator/file_generator.rb]: Fixed

File: lib/file_utils/file_generator/file_generator.rb (hunk 2 above)
Comment (slavapa) [inline lib/file_utils/file_generator/file_generator.rb:16]: I dropped one_kb string at all.

**Comment by kolmanv** [comment]
You have not removed the parameters file.
Please talk with Yaron and fix this Issue.
I will merge but only because this code is already in!
